src/Microsoft.DotNet.ProjectModel/CompilationOutputFiles.cs

```diff
@@ -58,14 +59,27 @@ public string PdbPath
 
         public virtual IEnumerable<string> Resources()
         {
-            var resourceNames = Project.Files.ResourceFiles
-                .Select(f => ResourceUtility.GetResourceCultureName(f.Key))
-                .Where(f => !string.IsNullOrEmpty(f))
-                .Distinct();
+            IEnumerable<string> resourceCultureNames = null;
+            var compilationOptions = Project.GetCompilerOptions(Framework, Configuration);
+            if (compilationOptions.EmbedInclude == null)
```

Comment:
How about you create a method to get the files and contain this if/else condition. It would do if() return Project.Files.ResourceFiles else { 
var resolver = new IncludeFilesResolver(compilationOptions.EmbedInclude);
return resolver.GetIncludeFiles("/")
}

The linq query is the same after that. I believe this would make the code cleaner.